### ISSUE TITLE
vanilla setup doesn't enable type aware linting

### ISSUE THREAD

**Issue by jemhuntr** [sveltejs/eslint-plugin-svelte3#144]
Global types and custom declarations aren't being detected during lint.

```ts
// src/global.d.ts
/// <reference types="@sveltejs/kit" />

declare interface CustomGlobalType {
	id: number;
	name: string;
}
```

```svelte
<!-- src/index.svelte -->
<script lang="ts">
	const onClick: svelte.JSX.MouseEventHandler<HTMLButtonElement> = (evt) => {
		const object: CustomGlobalType = { id: 0, name: evt.currentTarget.innerText };
		alert(JSON.stringify(object));
	};
</script>

<button on:click={onClick}>Click Me</button>
```

Running `npm run lint` yields the following errors:

![image](https://user-images.githubusercontent.com/6434778/142341886-ac634df2-a8ee-43f1-abd9-a274ca202f62.png)

Repro: https://github.com/jemhuntr/svelte-type-aware-eslint

---

Project has been created with the most minimal setup possible.

```
npm init svelte@next svelte-type-aware-eslint
√ Which Svelte app template? » Skeleton project
√ Use TypeScript? ... No / Yes
√ Add ESLint for code linting? ... No / Yes
√ Add Prettier for code formatting? ... No / Yes

npm install --save-dev eslint-plugin-svelte3
```

Modified eslintrc based on README documentation, and made some small formatting/linting tweaks to fix lint errors ootb.

**Comment by bamadesigner** [comment]
I'm currently having the same exact issue. Would love to know if anyone has figured this out.
